unstable/wlr-data-control-unstable-v1.xml

```diff
@@ -183,6 +183,74 @@
       <entry name="used_source" value="1"
         summary="source given to set_selection or set_primary_selection was already used before"/>
     </enum>
+
+    <!-- Version 3 additions -->
+
+    <event name="selection_serial" since="3">
+      <description summary="copy data to the selection">
+        This should be sent by compositors before a selection event with an
+        increasing serial number.
+
+        This can be used by the client in partnership to set_selection_response to indicate
+        pairing of a selection event to a reply.
+      </description>
+      <arg name="serial" type="uint">
+    </event>
+
+    <request name="set_selection_response" since="3">
```

Comment:
Bikeshedding: maybe we could find a better name? Would `swap_selection` make sense, since it replaces the selection with the provided serial? Or `set_selection_with_serial`?

Reply:
Generally speaking in my use case I'm trying to go from being empty to having content. So "swap" seems slightly off.

set_selection_with_serial is a very literal approach but that works for me.

Reply:
Throwing out another alternative - just so we consider all the options. We could add the response serial to the data control source. It would be an optional thing to send there and avoid having two requests here.

Reply:
That sounds like a good idea! Sources cannot be re-used anyways.

Reply:
@davidedmundson, would you be willing to update the implementations if I update the protocol here? Would be nice to move forward with this PR. :)

Reply:
@davidedmundson ping, or can you directly update the protocol here too?

Reply:
>would you be willing to update the implementations if I update the protocol here? Would be nice to move forward with this PR. :)

I can update this too. Apologies for the delay. It's now in my immediate TODO queue.